flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/ReconciliationUtils.java

```diff
@@ -62,14 +66,20 @@ void updateForSpecReconciliationSuccess(
 
         var clonedSpec = ReconciliationUtils.clone(spec);
         AbstractFlinkSpec lastReconciledSpec = reconciliationStatus.deserializeLastReconciledSpec();
+        boolean upgrading = false;
         if (lastReconciledSpec != null && lastReconciledSpec.getJob() != null) {
             Long oldSavepointTriggerNonce = lastReconciledSpec.getJob().getSavepointTriggerNonce();
             clonedSpec.getJob().setSavepointTriggerNonce(oldSavepointTriggerNonce);
+            if (clonedSpec.getJob().getState() == JobState.RUNNING
```

Comment:
What about add a method to judge whether there is on-going upgrading ?

Reply:
I would prefer to keep it this way for now as it's only a simple enum check